yt/frontends/parthenon/io.py

```diff
@@ -0,0 +1,107 @@
+from itertools import groupby
+
+import numpy as np
+
+from yt.utilities.io_handler import BaseIOHandler
+from yt.utilities.logger import ytLogger as mylog
+
+
+# http://stackoverflow.com/questions/2361945/detecting-consecutive-integers-in-a-list
+def grid_sequences(grids):
+    g_iter = sorted(grids, key=lambda g: g.id)
+    for _, g in groupby(enumerate(g_iter), lambda i_x1: i_x1[0] - i_x1[1].id):
+        seq = [v[1] for v in g]
+        yield seq
+
+
+ii = [0, 1, 0, 1, 0, 1, 0, 1]
+jj = [0, 0, 1, 1, 0, 0, 1, 1]
+kk = [0, 0, 0, 0, 1, 1, 1, 1]
+
+
+class IOHandlerParthenon(BaseIOHandler):
+    _particle_reader = False
+    _dataset_type = "parthenon"
+
+    def __init__(self, ds):
+        super().__init__(ds)
+        self._handle = ds._handle
+
+    def _read_particles(
+        self, fields_to_read, type, args, grid_list, count_list, conv_factors
+    ):
+        pass
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+        chunks = list(chunks)
+        if any((ftype != "parthenon" for ftype, fname in fields)):
+            raise NotImplementedError
+        f = self._handle
+        rv = {}
+        for field in fields:
+            # Always use *native* 64-bit float.
+            rv[field] = np.empty(size, dtype="=f8")
+        ng = sum(len(c.objs) for c in chunks)
+        mylog.debug(
+            "Reading %s cells of %s fields in %s blocks",
+            size,
+            [f2 for f1, f2 in fields],
+            ng,
+        )
+        last_dname = None
+        for field in fields:
+            ftype, fname = field
+            dname, fdi = self.ds._field_map[fname]
+            if dname != last_dname:
+                ds = f[f"/{dname}"]
+            ind = 0
+            for chunk in chunks:
+                if self.ds.logarithmic:
+                    for mesh in chunk.objs:
+                        nx, ny, nz = mesh.mesh_dims // self.ds.index.mesh_factors
+                        data = np.empty(mesh.mesh_dims, dtype="=f8")
+                        for n, id in enumerate(mesh.mesh_blocks):
+                            data[
+                                ii[n] * nx : (ii[n] + 1) * nx,
+                                jj[n] * ny : (jj[n] + 1) * ny,
+                                kk[n] * nz : (kk[n] + 1) * nz,
+                            ] = ds[id, fdi, :, :, :].transpose()
+                        ind += mesh.select(selector, data, rv[field], ind)  # caches
+                else:
+                    for gs in grid_sequences(chunk.objs):
+                        start = gs[0].id - gs[0]._id_offset
+                        end = gs[-1].id - gs[-1]._id_offset + 1
+                        # Deprecate this fallback on next major Parthenon release
```

Comment:
Can you clarify what "next" means in this comment ? What's the reference version ?
Or maybe we don't need to deprecate the fallback at all actually, it seems like a reasonable amount of legacy code to carry around, but it'd still be nice if the comment indicated which versions of Parthenon it is for.

Reply:
This does not apply any more as the old version has been deprecated for well over a year (probably two) now.
I removed the conditional and raising an error in `data_structure.py` if the version could not be determined.